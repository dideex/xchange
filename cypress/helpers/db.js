const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
