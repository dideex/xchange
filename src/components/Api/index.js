import {noty} from '../common'

export class Api {
  constructor() {
    this.backend = 'http://localhost:3030'
    this.isNetworkError = false
  }

  post(endpoint, data = {}, token = null) {
    return fetch(`${this.backend}/api/${endpoint}`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
      body: JSON.stringify(data),
    }).then(response => response.json())
  }

  get(endpoint, params = '', token = null) {
    return fetch(`${this.backend}/api/${endpoint}${params}`, {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
    }).then(response => response.json())
  }
  
  errorEmitter = fn => ({err, errCode, ...data}) => {
    if (err) {
      this.isNetworkError = true
      console.error('errorEmitter ___ ', errCode)
      noty(err, 'error')
    } else fn(data)
  }
}

export default new Api()
