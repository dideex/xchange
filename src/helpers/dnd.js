import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
