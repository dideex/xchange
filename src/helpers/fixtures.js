export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}