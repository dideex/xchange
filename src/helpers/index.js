export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
