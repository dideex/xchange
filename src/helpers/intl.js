import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
