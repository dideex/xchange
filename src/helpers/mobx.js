import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
