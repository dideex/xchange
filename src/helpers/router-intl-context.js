import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
