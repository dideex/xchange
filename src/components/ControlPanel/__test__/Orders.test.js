import React from 'react'
import Component from '../Orders'
import {shallow} from 'enzyme'
import {Provider as MobxProvider} from 'mobx-react'

import Api from '../../../components/Api'

import UserStore from '../../../store/User'

import {mountWrap, testId} from '../../../helpers/router-intl-context'
import {fakeData} from '../../../helpers/fixtures'

jest.mock('../../../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}, eth: {}}),
  errorEmitter: fn => data => fn(data),
}))

jest.mock('../../../components/common/Noty.js', () => ({
  noty: () => {},
}))

describe('Control panel: order', () => {
  let userStore
  beforeEach(() => {
    userStore = new UserStore()
    userStore.fetchGuestOrder = () => Promise.resolve({...fakeData, paymentStatus: 1})
  })

  describe('Component markup', () => {
    it('Should render without existing orders', () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  
})
