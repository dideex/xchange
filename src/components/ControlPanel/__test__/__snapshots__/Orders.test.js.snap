// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control panel: order Component markup Should render without existing orders 1`] = `"<div class=\\"css-iouif8\\"><h3><span>You have not yet made a transfer</span></h3></div>"`;
